src/Microsoft.Azure.WebJobs.Protocols/HeartbeatMonitor.cs

```diff
@@ -16,14 +17,13 @@ public class HeartbeatMonitor : IHeartbeatMonitor
 
         /// <summary>Initializes a new instance of the <see cref="HeartbeatMonitor"/> class.</summary>
         /// <param name="client">A blob client for the storage account in which to monitor heartbeats.</param>
-        
         public HeartbeatMonitor(CloudBlobClient client)
         {
             _client = client;
         }
 
         /// <inheritdoc />
-        public DateTimeOffset? GetSharedHeartbeatExpiration(string sharedContainerName, string sharedDirectoryName,
+        public async Task<DateTimeOffset?> GetSharedHeartbeatExpirationAsync(string sharedContainerName, string sharedDirectoryName,
```

Comment:
The Dashboard references these protocol shared types. If we've removed all the Dashboard code it will now diverge. Since we need to continue to support the Dashboard for the foreseeable future, how can we handle this?